source/blog/2016-12-06-part1-install-origin-on-f25-atomic-host.html.md

```diff
@@ -0,0 +1,349 @@
+author: dustymabe
+date: 2016-12-06 15:23:32 UTC
+tags: fedora, atomic, origin, openshift, kubernetes
+published: true
+comments: true
+---
+
+Introduction
+============
+
+[Openshift Origin](https://github.com/openshift/origin) is the upstream
+project that builds on top of the Kubernetes platform and feeds into the
+OpenShift Container Platform product that is available from Red Hat today.
+Origin is a great way to get started with Kubernetes and what better
+place to run a container orchestration layer than on top of Fedora
+Atomic Host!
+
+We recently released [Fedora
+25](https://fedoramagazine.org/fedora-25-released/) and along with that
+was the first two week release of Fedora 25 Atomic Host. This blog post
+will show you the basics for getting a production installation of Origin
+running on Fedora 25 Atomic Host using the [OpenShift Ansible
+Installer](https://github.com/openshift/openshift-ansible).
+
+This first post will cover just the installation. In a later blog post
+we'll take the system we just installed for a spin and make sure
+everything is working as expected.
+
+Environment
+===========
+
+I've tried to make this setup as generic as possible. In this case I
+will be targeting three generic servers that are running Fedora 25
+Atomic Host. As is common with cloud environments these servers each
+have an "internal" private address that can't be accessed from the
+internet, as well as a public NATed address that can be accessed from
+the outside. Here is the identifying information for the three servers:
+
+|    Role     |  Public IPv4   | Private IPv4 |
+|-------------|----------------|--------------|
+| master,etcd | 54.175.0.44    | 10.0.173.101 |
+| worker      | 52.91.115.81   | 10.0.156.20  |
+| worker      | 54.204.208.138 | 10.0.251.101 |
+
+As you can see from the table I've marked one of the nodes as the master
+and the other two as what I'm calling *worker nodes*. The master node
+will run the api server, scheduler and, controller manager. We'll also
+run etcd on it. Since we want to make sure we don't starve the node running
+etcd, we'll mark the master node as **unschedulable** so that
+application containers don't get scheduled to run on it.
+
+The other two nodes, the worker nodes, will have the proxy and the
+kubelet running on them; this is where the containers (inside of pods)
+will get scheduled to run. We'll also tell the installer to run a
+registry and an HAProxy router on the two worker nodes so that we can
+perform builds as well as access our services from the outside world via
+HAProxy.
+
+The Installer
+=============
+
+[Openshift Origin](https://github.com/openshift/origin) uses
+[Ansible](https://www.ansible.com/) to manage the installation of
+different nodes in a cluster. The code for this is aggregated in the
+[OpenShift Ansible
+Installer](https://github.com/openshift/openshift-ansible) on GitHub.
+We'll start by cloning the repo:
```

Comment:
Why don't we offer a container for this?  It would make life much easier for our users.